src/frame/mod.rs

```diff
@@ -271,6 +392,30 @@ pub struct ExceptionResponse {
     pub exception: Exception,
 }
 
+/// Convenience trait for downcasting `std::io::Error` to `ExceptionResponse`
```

Comment:
Why do we need to introduce a public trait? What is the use case?

Reply:
I've added an example to make this more clear. 
Just an easy way to downcast the std::io::Error to the ExceptionResponse if it is one. 
I'm also not really with this solution. The better way would be to introduce an Error type for the clients. Could look like this

```rust
pub enum Error {
    ModbusException(ModbusException),
    IoError(std::io::Error)
}
```

This this change it's clear for the api users that a ModbusException exists and can be extracted. On the other hand that could break a lot of builds. 
What do you think?